Allow enabling websocket compression in NodeHttpServer and BunHttpServer (#6691)
Effect-TS/effect: .changeset/http-server-websocket-options.md

---
"@effect/platform-node": patch
"@effect/platform-bun": patch
---

Allow configuring the WebSocket server in `NodeHttpServer` and `BunHttpServer`.

> The HTTP server layers previously hardcoded their WebSocket wiring — `NodeHttpServer` always constructed `new WebSocketServer({ noServer: true })`, and `BunHttpServer` overwrote any `websocket` key in the serve options with its own handlers. There was no way to enable `permessage-deflate` compression (which browsers offer on every connection) or tune payload limits.
> 
> Both servers now accept a `websocket` option that is forwarded to the underlying implementation, with the wiring/lifecycle options the server manages excluded from the type:

Reviewer: ```suggestion
Allow configuring the WebSocket server in `NodeHttpServer` and `BunHttpServer`.

Both servers now accept a `websocket` option that is forwarded to the underlying implementation, with the wiring/lifecycle options the server manages excluded from the type:
```

Author: Applied in 51c31fb — changeset trimmed to the suggested copy.
